tests/integration/kubernetes/k8s-confidential.bats

```diff
@@ -10,12 +10,17 @@ load "${BATS_TEST_DIRNAME}/confidential_common.sh"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
 
 setup() {
-	SUPPORTED_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+	SUPPORTED_TEE_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+	SUPPORTED_NON_TEE_HYPERVISORS=("qemu")
 
 	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
 	# having substrings, like qemu, being matched with qemu-$something.
-	[[ " ${SUPPORTED_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]] ||  skip "Test not supported for ${KATA_HYPERVISOR}."
+	if ! [[ "${SUPPORTED_TEE_HYPERVISORS[@]}" =~ "${KATA_HYPERVISOR}" ]] && [[ "${SUPPORTED_NON_TEE_HYPERVISORS}" =~ "${KATA_HYPERVISOR}" ]]; then
```

Comment:
```suggestion
	if ! [[ "${SUPPORTED_TEE_HYPERVISORS[@]}" =~ "${KATA_HYPERVISOR}" ]] && ! [[ "${SUPPORTED_NON_TEE_HYPERVISORS}" =~ "${KATA_HYPERVISOR}" ]]; then
```

Otherwise the logic is not correct.